### PR TITLE
13848 fix mhr table child ux issues

### PR DESCRIPTION
**Pull request by ChasBoucher**
*Issue #:* /bcgov/entity#13848

*Description of changes:*

- Add Toast message, auto scroll when submitting(Review and Confirm)
- Includes #14398 - a very small fix to allow for Submitting Drafts

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the PPR license (Apache 2.0).

### CODE REVIEW

File: ppr-ui/src/utils/feature-flags.ts [hunk 1]
```diff
@@ -9,8 +9,8 @@ export const defaultFlagSet: LDFlagSet = {
   'bcregistry-ui-mhr-enabled': false,
   'search-registration-number': false,
   'search-serial-number': false,
-  'mhr-ui-enabled': false,
-  'mhr-registration-enabled': false
+  'mhr-ui-enabled': true,
```
Comment (dimak1) [inline ppr-ui/src/utils/feature-flags.ts]: Undo these.

File: ppr-ui/src/views/mhrInformation/MhrInformation.vue [hunk 2]
```diff
@@ -408,6 +408,13 @@ export default defineComponent({
           setUnsavedChanges(false)
           // Delete the draft on successful submission
           if (getMhrInformation.value.draftNumber) await deleteMhrDraft(getMhrInformation.value.draftNumber)
+          const newItem: RegTableNewItemI = {
+            addedReg: String(getMhrInformation.value.mhrNumber),
```
Comment (dimak1) [inline ppr-ui/src/views/mhrInformation/MhrInformation.vue]: What about using `.toString()` ?

File: ppr-ui/src/views/mhrInformation/MhrInformation.vue [hunk 3]
```diff
@@ -408,6 +408,13 @@ export default defineComponent({
...
+            addedRegParent: '',
+            addedRegSummary: null,
+            prevDraft: String(getMhrInformation.value.documentId) || ''
```
Comment (dimak1) [inline ppr-ui/src/views/mhrInformation/MhrInformation.vue]: Same as above.

File: ppr-ui/src/composables/mhrInformation/useMhrInformation.ts [hunk 4]
```diff
@@ -95,7 +95,7 @@ export const useMhrInformation = () => {
       declaredValue: getMhrTransferDeclaredValue.value,
       consideration: getMhrTransferConsideration.value,
       transferDate: getMhrTransferDate.value,
-      ownLand: getMhrTransferOwnLand.value,
+      ownLand: getMhrTransferOwnLand.value === null ? false : getMhrTransferOwnLand.value,
```
Comment (dimak1) [inline ppr-ui/src/composables/mhrInformation/useMhrInformation.ts]: `ownLand: getMhrTransferOwnLand.value || false`

File: ppr-ui/src/components/mhrTransfers/TransferDetails.vue [hunk 5]
```diff
@@ -180,7 +180,7 @@ export default defineComponent({
       declaredValue: getMhrTransferDeclaredValue.value?.toString(),
       consideration: getMhrTransferConsideration.value,
       transferDate: getMhrTransferDate.value,
-      isOwnLand: getMhrTransferOwnLand.value,
+      isOwnLand: getMhrTransferOwnLand.value === null ? false : getMhrTransferOwnLand.value,
```
Comment (dimak1) [inline ppr-ui/src/components/mhrTransfers/TransferDetails.vue]: `isOwnLand: getMhrTransferOwnLand.value || false`

**Review by dimak1 — CHANGES_REQUESTED** [review]

File: ppr-ui/src/components/mhrTransfers/TransferDetails.vue (hunk 5 above)
Comment (ChasBoucher) [reply ppr-ui/src/components/mhrTransfers/TransferDetails.vue]: ok

File: ppr-ui/src/composables/mhrInformation/useMhrInformation.ts (hunk 4 above)
Comment (ChasBoucher) [reply ppr-ui/src/composables/mhrInformation/useMhrInformation.ts]: ok

File: ppr-ui/src/views/mhrInformation/MhrInformation.vue (hunk 2 above)
Comment (ChasBoucher) [reply ppr-ui/src/views/mhrInformation/MhrInformation.vue]: good point...

File: ppr-ui/src/views/mhrInformation/MhrInformation.vue (hunk 3 above)
Comment (ChasBoucher) [reply ppr-ui/src/views/mhrInformation/MhrInformation.vue]: done..

File: ppr-ui/src/utils/feature-flags.ts (hunk 1 above)
Comment (ChasBoucher) [reply ppr-ui/src/utils/feature-flags.ts]: oops

File: ppr-ui/src/views/mhrInformation/MhrInformation.vue (hunk 2 above)
Comment (ChasBoucher) [reply ppr-ui/src/views/mhrInformation/MhrInformation.vue]: I did some testing with .toString() and it fails... 

Looked up the reason why and String(x) ensures the value of the object converted.
So I tried value.toString() and get a type error... I think String(x) is the most reliable...

File: ppr-ui/src/views/mhrInformation/MhrInformation.vue (hunk 3 above)
Comment (ChasBoucher) [reply ppr-ui/src/views/mhrInformation/MhrInformation.vue]: Same as above

File: ppr-ui/src/components/mhrTransfers/TransferDetails.vue (hunk 5 above)
Comment (dimak1) [reply ppr-ui/src/components/mhrTransfers/TransferDetails.vue]: Missed this change.

**Review by dimak1 — CHANGES_REQUESTED** [review]
One more change that was missed.

File: ppr-ui/src/components/mhrTransfers/TransferDetails.vue (hunk 5 above)
Comment (ChasBoucher) [reply ppr-ui/src/components/mhrTransfers/TransferDetails.vue]: not sure how that happened... Anyways... I have fixed the issue.

File: ppr-ui/src/views/mhrInformation/MhrInformation.vue (hunk 2 above)
Comment (dimak1) [reply ppr-ui/src/views/mhrInformation/MhrInformation.vue]: Hmm interesting why is it failing on type error, as `addedReg: string` is a string type.

Comment (cameron-eyds) [reply ppr-ui/src/views/mhrInformation/MhrInformation.vue]: Why hard type cast it at all? 
isn't `mhrNumber` typed and initialized as a string?

Comment (cameron-eyds) [reply ppr-ui/src/views/mhrInformation/MhrInformation.vue]: `MhRegistrationSummaryIF`

Comment (ChasBoucher) [reply ppr-ui/src/views/mhrInformation/MhrInformation.vue]: do you want me to change it?

Comment (cameron-eyds) [reply ppr-ui/src/views/mhrInformation/MhrInformation.vue]: Is the code required/doing something or this is a decoy?

If the latter, then yes i'd just go with `addedReg:  getMhrInformation.value.mhrNumber`. Why cast it to a string if it already is, perhaps i'm missing something?

Comment (ChasBoucher) [reply ppr-ui/src/views/mhrInformation/MhrInformation.vue]: Did some testing without String and ran into an issue...  I think we can get away without it, but there is another issue...I am going to fix..

Comment (ChasBoucher) [reply ppr-ui/src/views/mhrInformation/MhrInformation.vue]: To answer your question..  This code is required for locating the current record saved.  I is used to ensure that the scrolling on the table locates the correct record...

Comment (cameron-eyds) [reply ppr-ui/src/views/mhrInformation/MhrInformation.vue]: The code I was referring to was the casting of the value, not assigning the property value.

You are saying the `String()` is required? 
Seems like an underlying issue if we need to hard cast a string to a string..

Comment (ChasBoucher) [reply ppr-ui/src/views/mhrInformation/MhrInformation.vue]: Made change..

Comment (cameron-eyds) [reply ppr-ui/src/views/mhrInformation/MhrInformation.vue]: Resolved 👍

**Review by cameron-eyds — COMMENTED** [review]
LGTM!

**Review by cameron-eyds — APPROVED** [review]
LGTM

File: ppr-ui/src/views/mhrInformation/MhrInformation.vue (hunk 3 above)
Comment (ChasBoucher) [reply ppr-ui/src/views/mhrInformation/MhrInformation.vue]: Made change to no conversion..  Both values are strings... Not sure why I did that in the first place.

**Review by doug-lovett — APPROVED** [review]

**Review by dimak1 — APPROVED** [review]

**Pull request merged**